scenarios/sslstream.benchmarks.yml

```
@@ -0,0 +1,95 @@
+# Example:
+# crank --config https://raw.githubusercontent.com/aspnet/Benchmarks/main/scenarios/sslstream.benchmarks.yml --scenario read-write --profile aspnet-perf-lin
+
+imports:
+  - https://raw.githubusercontent.com/aspnet/Benchmarks/main/scenarios/aspnet.profiles.standard.yml
+
+variables:
+  # common parameters
+  serverPort: 5000
+  sendBufferSize: 32768 # bytes
+  receiveBufferSize: 32768 # bytes
+  tlsVersion: "1.3" # <1.2|1.3>
+  serverCertSelection: "CertContext" # <Callback|Certificate|CertContext>
+  clientCertSelection: "CertContext" # <Callback|Collection|CertContext>
+
+  # client-specific parameters
+  warmup: 15 #seconds
+  duration: 15 #seconds
+
+  # server-specific parameters
+  # requireClientCert: false
+
+scenarios:
+  read-write:
+    server:
+      job: sslServer
+      agent: secondary
+      variables:
+        host: "{{secondaryAddress}}"
+    client:
+      job: sslClient
+      agent: main
+      variables:
+        scenario: ReadWrite
+        host: "{{secondaryAddress}}"
+
+  client-read:
+    server:
+      job: sslServer
+      agent: secondary
+      variables:
+        host: "{{secondaryAddress}}"
+        receiveBufferSize: 0
+    client:
+      job: sslClient
+      agent: main
+      variables:
+        scenario: ReadWrite
+        sendBufferSize: 0
+
+  client-write:
+    server:
+      job: sslServer
+      agent: secondary
+      variables:
+        host: "{{secondaryAddress}}"
+        sendBufferSize: 0
+    client:
+      job: sslClient
+      agent: main
+      variables:
+        scenario: ReadWrite
+        receiveBufferSize: 0
+
+  handshake:
+    server:
+      job: sslServer
+      agent: secondary
+      variables:
+        host: "{{secondaryAddress}}"
+    client:
+      job: sslClient
+      agent: main
+      variables:
+        scenario: Handshake
+        host: "{{secondaryAddress}}"
+
+jobs:
+  sslServer:
+    source:
+      repository: https://github.com/aspnet/benchmarks.git
+      branchOrCommit: main
+      project: src/BenchmarksApps/SslStream/Server/SslStreamServer.csproj
+    readyStateText: Listening on # app should write this line to output -- default Kestrel logging does that
+    arguments: "--port {{serverPort}} --receive-buffer-size {{receiveBufferSize}} --send-buffer-size {{sendBufferSize}} --tls-version {{tlsVersion}} --cert-selection {{serverCertSelection}}"
+
+  sslClient:
+    source:
+      repository: https://github.com/aspnet/benchmarks.git
+      branchOrCommit: main
+      project: src/BenchmarksApps/SslStream/Client/SslStreamClient.csproj
+    isConsoleApp: true
+    waitForExit: true
+    timeout: 300 #seconds
+    arguments: "--host {{secondaryAddress}} --port {{serverPort}} --receive-buffer-size {{receiveBufferSize}} --send-buffer-size {{sendBufferSize}} --tls-version {{tlsVersion}} --cert-selection {{clientCertSelection}} --scenario {{scenario}} --warmup 00:00:{{warmup}} --duration 00:00:{{duration}}"

```

hmmm I think the first one should be "host" instead of "secondaryAddress" 🤔 so that if you change the address, you change it in variables only

```suggestion
    arguments: "--host {{host}} --port {{serverPort}} --receive-buffer-size {{receiveBufferSize}} --send-buffer-size {{sendBufferSize}} --tls-version {{tlsVersion}} --cert-selection {{clientCertSelection}} --scenario {{scenario}} --warmup 00:00:{{warmup}} --duration 00:00:{{duration}}"

```